lib/insights/api/common/application_controller_mixins/request_body_validation.rb

```diff
@@ -18,11 +18,9 @@ def self.included(other)
 
           def body_params
             @body_params ||= begin
-              raw_body    = request.body.read
-              parsed_body = raw_body.blank? ? {} : JSON.parse(raw_body)
-              ActionController::Parameters.new(parsed_body).permit!
-            rescue JSON::ParserError
-              raise Insights::API::Common::ApplicationControllerMixins::RequestBodyValidation::BodyParseError, "Failed to parse request body, expected JSON"
+              hash = params.permit!.to_h
```

Comment:
What other content-types do you want to allow?  What happens when you call `.to_h` on them?

Reply:
@bdunne 
For catalog we need to support application/json and multipart/form-data as listed here in the openapi

https://swagger.io/docs/specification/describing-request-body/multipart-requests/